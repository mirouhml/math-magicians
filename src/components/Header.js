import React from 'react';
import {
  HashRouter as Router,
  Route,
  Routes,
  Link,
} from 'react-router-dom';
import Home from './HomePage';
import QuotesPage from './QuotesPage';
import CalculatorPage from './CalculatorPage';

const Header = () => (
  <Router>
    <div className="main">
      <div className="header">
        <h1>Math Magicians</h1>
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/calculator">Calculator</Link>
          </li>
          <li>
            <Link to="/quotes">Quotes</Link>
          </li>
        </ul>
      </div>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/calculator" element={<CalculatorPage />} />
        <Route path="/quotes" element={<QuotesPage />} />
      </Routes>
    </div>
  </Router>
);

export default Header;
