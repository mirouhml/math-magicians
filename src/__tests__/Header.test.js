import { screen, render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

describe('Header component tests:', () => {
  it('Header component renders correctly', () => {
    const tree = renderer
      .create(<Header />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Checking if the HomePage is the landing page', () => {
    const { container } = render(<Header />);
    const home = container.querySelector('.home');
    expect(home).toBeInTheDocument();
  });

  it('Switching the page then returning to the HomePage', () => {
    const { container } = render(<Header />);
    fireEvent.click(screen.getByText('Calculator'));
    fireEvent.click(screen.getByText('Home'));
    const home = container.querySelector('.home');
    expect(home).toBeInTheDocument();
  });

  it('Clicking on Calculator and getting the CalculatorPage', () => {
    const { container } = render(<Header />);
    fireEvent.click(screen.getByText('Calculator'));
    const page = container.querySelector('.calculator-page');
    expect(page).toBeInTheDocument();
  });

  it('Clicking on Quotes and getting the QuotesPage', () => {
    const { container } = render(<Header />);
    fireEvent.click(screen.getByText('Quotes'));
    const page = container.querySelector('.quotes');
    expect(page).toBeInTheDocument();
  });
});
