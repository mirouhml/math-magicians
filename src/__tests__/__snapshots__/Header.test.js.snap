// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests: Header component renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="header"
  >
    <h1>
      Math Magicians
    </h1>
    <ul>
      <li
        className="active-background"
      >
        <a
          href="#/"
          id="home"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="#/calculator"
          id="calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="#/quotes"
          id="quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </ul>
  </div>
  <div
    className="background home"
  >
    <div>
      <h2>
        Welcome to our page!
      </h2>
      <p>
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
      </p>
      <p>
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
      </p>
    </div>
  </div>
</div>
`;
